### ISSUE TITLE
Use tabpagenr('#') to get previous tabpagenr

### ISSUE THREAD

**Issue by wsdjeg** [vim/vim#6657]
**Is your feature request about something that is currently impossible or hard to do? Please describe the problem.**
I do not know if it is possible to get previous tabpagenr in viml.

**Describe the solution you'd like**
in neovim, we can use `tabpagenr('#')` to get previous tabpagenr

ref: https://github.com/SpaceVim/SpaceVim/issues/3688

**Comment by brammool** [comment]
What does "previous" mean here?  If you look at "gt" and :tabprevious it means the tab page on the left of the current one.
You could just use tabpagenr() - 1 for that.

**Comment by Sillyringo** [comment]
> What does "previous" mean here? If you look at "gt" and :tabprevious it means the tab page on the left of the current one.
> You could just use tabpagenr() - 1 for that.

I guess "previous_tab" means "the tab you use before", like tab_a when you switch from tab_a to tab_b

**Comment by chrisbra** [comment]
That should be possible using tablocal variables and TabEnter autocommands.

**Comment by wsdjeg** [comment]
@chrisbra Yeah, but when we use `noautocmd`, It will not be possible to set tablocal var.

**Comment by wsdjeg** [comment]
@brammool I mean last used tabpagenr. I also try to use `:tabs`, but it does not include `#` like `:ls`. in neovim, we can use `tabpagenr('#')` or parser output of `:tabs`. It looks like:

```


Tab page 1
    ~\.SpaceVim.d\init.toml
    [defx] -0
Tab page 2
#   ~\.SpaceVim.d\init.toml
Tab page 3
>   ~\.SpaceVim.d\init.toml
```

**Comment by lacygoill** [comment]
Possibly fixed by [8.2.1401](https://github.com/vim/vim/commit/62a232506d06f6d1b3b7271801c907d6294dfe84).

---

This todo item might no longer be relevant:
https://github.com/vim/vim/blob/127542bcebeb6480493b09d75a3be1d98a5f7797/runtime/doc/todo.txt#L2866-L2867

**Comment by wsdjeg** [comment]
wow, Thanks.
